promoter/database.go

```diff
@@ -362,6 +365,26 @@ OUTER:
 	}
 }
 
+// threadedPruneLocks periodically scans the db for prunable locks.
+func (p *Promoter) threadedPruneLocks() {
+	t := time.NewTicker(lockPruningInterval)
+	defer t.Stop()
+
+	purger := lock.NewPurger(p.staticLockClient)
+	for range t.C {
+		select {
+		case <-p.staticBGCtx.Done():
```

Comment:
The pruning interval is 24 hours and we're ranging on its ticker.
We should have a conditionless loop where we block on the ticker and the bgctx without a default.